rest/src/main/java/io/syndesis/rest/v1/handler/integration/IntegrationHandler.java

```diff
@@ -108,6 +121,44 @@ public Integration get(String id) {
         return integration;
     }
 
+
+    @GET
+    @Path("/{id}/export.zip")
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public StreamingOutput export(@NotNull @PathParam("id") @ApiParam(required = true) String id) throws IOException {
+        ArrayList<ModelData> models = new ArrayList<>();
+
+        Integration integration = this.get(id);
+        models.add(new ModelData(Kind.Integration, integration));
+
+        for (Step step : integration.getSteps()) {
+            Optional<Connection> c = step.getConnection();
+            if( c.isPresent() ) {
+                Connection connection = c.get();
+                models.add(new ModelData(Kind.Connection, connection));
+                Connector connector = getDataManager().fetch(Connector.class, connection.getConnectorId().get());
+                if( connector != null ) {
+                    models.add(new ModelData(Kind.Connector, connector));
+                }
+            }
+        }
+
+        return out -> {
+            try (ZipOutputStream tos = new ZipOutputStream(out) ) {
+                addEntry(tos, EXPORT_MODEL_FILE_NAME, Json.mapper().writeValueAsBytes(models));
```

Comment:
Would a ZIP file with multiple entries make more sense? the file names could be sanitized names of Integrations/Connectors/Connections?

Reply:
For the moment I don't expect many people looking into the zip so its somewhat opaque. 

What I miss though is some sort of versioning, which allows for evolving the export/import format over time.

Reply:
I agree, but I think we also need this in the DB model for the same reasons.  So I think once we put that in place there, this export/import thing will likely pick it up for free.